### PR TITLE
Use test and testing feature flag instead of unused when needed.

### PR DESCRIPTION
**Pull request by AureliaDolo**
The idea is to hide behind compilation flags the functions that are used only in tests.

### CODE REVIEW

**Comment by AureliaDolo** [comment]
Also there is a lot of `#[allow(unused)]`  and `#[allow(dead_code)]` without specifying why so maybe it could be worth having a closer look at that :eyes:

**Comment by AureliaDolo** [comment]
So I kinda fell into a rabbit hole, some "only used in tests" functions are indeed not used.

File: libparsec/crates/platform_storage/Cargo.toml [hunk 1]
```diff
@@ -39,6 +41,8 @@ serde-wasm-bindgen = { workspace = true }
 libparsec_tests_lite = { workspace = true }
 # Note `libparsec_tests_fixtures` enables our `test-with-testbed` feature
 libparsec_tests_fixtures = { workspace = true, default-features = false, features = ["test-with-platform-storage-testbed"] }
+libparsec_platform_storage = {path = ".",features = ["testing"]}
```
Comment (vxgmichel) [inline libparsec/crates/platform_storage/Cargo.toml:44]: It's not formatted like the line above.

**Review by vxgmichel — COMMENTED** [review]
I don't understand why we sometimes have:
```rust
    #[cfg(any(test, feature = "testing"))]
    #[allow(unused)]
```
or only this:
```rust
#[cfg(any(test, feature = "testing"))]
```
or even that:
```rust
#[cfg(feature = "testing")]
```

Could you provide a bit more info about that?

**Comment by AureliaDolo** [comment]
@vxgmichel 

plateform_storage is a dependency of the client. And some `debug_dump` are used in the client so we cannot just use `#[cfg(test)]` there. 

So to detail on the cases you mentionned:

```rust
    #[cfg(any(test, feature = "testing"))]
    #[allow(unused)]
```

This is because the function is *really* never used; I choose to add both flags in case it is used in the future

```rust
#[cfg(any(test, feature = "testing"))]
```

This is when the function is used both in its crate tests and in another crate's tests

```rust
#[cfg(feature = "testing")]
```
And this is when the function is only used in another crate's tests.

Could you provide a bit more info about that?

**Comment by vxgmichel** [comment]
> This is because the function is _really_ never used; I choose to add both flags in case it is used in the future

Then maybe use `#[allow(unused)]` only and replace it with `#[cfg(...)]` once it's used?

> This is when the function is used both in its crate tests and in another crate's tests

Is there a way to merge `test` and `testing` somehow? I find it a bit confusing. Or maybe improve the wording.

**Review by vxgmichel — APPROVED** [review]
See my previous comment, I don't have a strong opinion though.

LGTM anyway :+1:

**Comment by AureliaDolo** [comment]
> Then maybe use `#[allow(unused)]` only and replace it with `#[cfg(...)]` once it's used?

If we do this, there won't be any warning if someone uses it not in tests

> Is there a way to merge test and testing somehow? I find it a bit confusing. Or maybe improve the wording.

I don't think so. The `test` attribute is part of the language whereas `testing` is a feature that's defined in the project and we could choose to expose anything behind it. Maybe `test_interface` ?

**Comment by vxgmichel** [comment]
> Maybe `test_interface` ?

Or `expose-test-methods`? I'll let you choose what you find less confusing.

**Review by FirelightFlagboy — APPROVED** [review]
LGTM,

concerning the naming of the features, we could keep `testing` or I would go with the followings:

- `expose-test-methods`
- `expose-debug-methods`

Since rust features should be additive:

- just `test-methods`
- just `debug-methods`

> I understand that as _add {tests,debug} methods_

File: libparsec/crates/platform_storage/Cargo.toml [hunk 2]
```diff
@@ -39,6 +41,8 @@ serde-wasm-bindgen = { workspace = true }
 libparsec_tests_lite = { workspace = true }
 # Note `libparsec_tests_fixtures` enables our `test-with-testbed` feature
 libparsec_tests_fixtures = { workspace = true, default-features = false, features = ["test-with-platform-storage-testbed"] }
+libparsec_platform_storage = {path = ".",features = ["expose-test-methods"]}
 
 [target.'cfg(target_arch = "wasm32")'.dev-dependencies]
 wasm-bindgen-test = { workspace = true }
+libparsec_platform_storage = {path = ".",features = ["expose-test-methods"]}
```
Comment (FirelightFlagboy) [inline libparsec/crates/platform_storage/Cargo.toml:48]: The first include in `dev-dependencies` is not enough ?

Comment (AureliaDolo) [reply libparsec/crates/platform_storage/Cargo.toml]: Indeed

File: libparsec/crates/platform_storage/Cargo.toml [hunk 3]
```diff
@@ -39,6 +41,7 @@ serde-wasm-bindgen = { workspace = true }
 libparsec_tests_lite = { workspace = true }
 # Note `libparsec_tests_fixtures` enables our `test-with-testbed` feature
 libparsec_tests_fixtures = { workspace = true, default-features = false, features = ["test-with-platform-storage-testbed"] }
+libparsec_platform_storage = {path = ".",features = ["expose-test-methods"]}
```
Comment (FirelightFlagboy) [inline libparsec/crates/platform_storage/Cargo.toml:44]: Missing space around brace and after comma:

```suggestion
libparsec_platform_storage = { path = ".", features = ["expose-test-methods"] }
```

**Review by FirelightFlagboy — APPROVED** [review]
LGTM with a minor change

**Pull request merged**